plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -2121,6 +2137,210 @@ var _ = Describe("validator", func() {
 					Expect(err.Error()).To(ContainSubstring("spec.provider.workers[2].volume.size"))
 				})
 			})
+
+			Context("RawExtension internal API usage checks", func() {
+				BeforeEach(func() {
+					shoot.Spec.Provider.InfrastructureConfig = &runtime.RawExtension{
+						Raw: []byte(`{
+"kind": "InfrastructureConfig",
+"apiVersion": "azure.provider.extensions.gardener.cloud/__internal",
+"networks": {"vnet": {"cidr": "10.250.0.0/16"}}
```

Comment:
Do you really need the content of `InfrastructureConfig` (and other kinds) to be so elaborate to make these tests pass or fail? I would shorten this as much as possible (or eliminate it completely).
```suggestion
"networks": {}
```

Reply:
You're right, this isn't necessary. I wanted to use as much of a real-life example as possible, because all the other tests I was looking at there were using random data, which probably caused me to err on the other side of the extreme here. It is probably enough to reduce this to examples with `kind`, `apiVersion` and at least one `"key": "value"` pair?

Reply:
Yes, that would be fine from my perspective.

Reply:
done.